src/ray/util/process.h

```diff
@@ -73,14 +74,18 @@ class Process {
   /// \param[in] decouple True iff the parent will not wait for the child to exit.
   /// \param[in] env Additional environment variables to be set on this process besides
   /// the environment variables of the parent process.
-  explicit Process(const char *argv[], void *io_service, std::error_code &ec,
-                   bool decouple = false, const ProcessEnvironment &env = {});
+  explicit Process(StartupToken token, const char *argv[], void *io_service,
+                   std::error_code &ec, bool decouple = false,
+                   const ProcessEnvironment &env = {});
   /// Convenience function to run the given command line and wait for it to finish.
   static std::error_code Call(const std::vector<std::string> &args,
                               const ProcessEnvironment &env = {});
   static Process CreateNewDummy();
-  static Process FromPid(pid_t pid);
-  pid_t GetId() const;
+  static Process FromPidAndToken(pid_t pid, StartupToken token);
+
+  StartupToken GetId() const;
```

Comment:
@wuisawesome What do you think about moving StartupToken to Process so that each Process has a ray generated unique id.

Reply:
I don't really mind. I think it actually makes sense to add here, but rename `Process` to `WorkerDescriptor` 

Also nit: can we keep calling it `GetStartupToken` instead of id, since the pid vs id vs etc is very confusing.

Reply:
I'm actually not sure about renaming Process to WorkerDescriptor since it's util so technically you can use it to start any processes not just workers.

Reply:
I guess my opinion here is that the StartupToken belongs with whatever is tracking the "worker process" (or whatever process group that is). 

It's a valid point though that this doesn't generalize to other process types. Maybe this means we should put it in `worker.h` (and pass worker classes around instead of processes within the worker pool), maybe we can just move the class into the worker pool in addition to renaming it (i believe it's only used for this one purpose). 

>  since there are places relying on GetId() returning pid

Isn't this a recipe for failure since that pid may be wrong?

Reply:
> Isn't this a recipe for failure since that pid may be wrong?

Yea, we need to look into those usages as well. Just leave it as it is to limit the scope of this PR.

I feel like both Process and StartupToken are generic and can be used whenever someone wants to spawn a process. StartupToken may sound not generic but it's just a unique id generated by ray for each process.

Reply:
Maybe we should change the name of `StartupToken` to `UniqueWorkerId`? I now understand that at least for the needs of `worker_pool.cc`, this token should be part of the worker class, which probably will solve the immediate problem of the `starting_workers_to_tasks` and `starting_dedicated_workers_to_tasks` maps.

Reply:
Just realized that @stephanie-wang has a PR that completely removed the `starting_workers_to_tasks` and `starting_dedicated_workers_to_tasks` maps so we won't have the problem.

@stephanie-wang Do you have a timeline to resubmit the PR? If it's soon then we can probably wait or we can have some short term fix.